src/third_party/wiredtiger/src/meta/meta_ckpt.c

```diff
@@ -240,6 +243,246 @@ __ckpt_compare_order(const void *a, const void *b)
 	return (ackpt->order > bckpt->order ? 1 : -1);
 }
 
+/*
+ * __ckpt_str_find_not_alphanum --
+ *  Return the first character that is not alphanumeric.
+ */
+const char *
+__ckpt_str_find_not_alphanum(const char *s)
+{
+	while (*s != '\0' && __wt_isalnum(*s))
+		s++;
+
+	return (s);
+}
+
+/*
+ * __ckpt_parse_attribute --
+ *
+ *  Look for the "prefix" provided starting from location "*s". If the prefix
+ *  is found, copy consecutive alphanumeric characters found after the prefix
+ *  into the buffer "val" and set "len" to the length of the value of the
+ *  attribute. "*s" is advanced past the end of the attribute.
+ *
+ *  If the prefix is not found, return WT_ERROR.
+ */
+static int
+__ckpt_parse_attribute(
+	const char *prefix, const char **s, char *val, int *len)
+{
+	const char *str_p = *s;
+	const char *prefix_p = prefix;
+
+	/*
+	 * While we have not reached the end of the prefix string or the search
+	 * string, and while we have not exited the current level of
+	 * configuration indicated by parens, look for a prefix match.
+	 */
+	while (*str_p != '\0' && *str_p != ')' && *prefix_p != '\0') {
```

Comment:
I wonder if this would be easier with strtok or strpbrk. Have you looked at those ?

Reply:
Hm yea that might be less code. I'll try it and see if it's similar in speed.

Reply:
I rewrote it with the library functions. Performance is more or less the same (slower by 60ms after 30 minutes).